### PR TITLE
Extend setup database create dump

### PR DESCRIPTION
**Pull request by vmcj**
This is the first half of https://github.com/DOMjudge/domjudge/pull/2180 but due to the amount of comments and needed tests I prefer to already merge the good parts.

This should create the same structure as the file we already had so together with `load-db` from the scripts repo this should work.

### CODE REVIEW

**Comment by eldering** [comment]
I don't think that taking parts out of https://github.com/DOMjudge/domjudge/pull/2180 helps: now all the comments are lost and it's more difficult to track what's going on where.

**Comment by vmcj** [comment]
> I don't think that taking parts out of #2180 helps: now all the comments are lost and it's more difficult to track what's going on where.

Those should be all the handled parts (and only the simple command). I think the last remarks were about the behaviour of loading the dump and that seems to be a discussion on itself (but can take place in the other PR). But if you still find issues with this PR I'm fine with handling it within the other one but for me it's a lot harder if I need to handle multiple things and keep on testing.

**Comment by vmcj** [comment]
@eldering as discussed yesterday, I prefer to already merge this part and then address the last issues on the `load` behaviour.

File: sql/dj_setup_database.in [hunk 1]
```diff
@@ -34,6 +35,7 @@ Commands:
   install-examples install examples only
   install-loadtest configure for load testing. WARNING: CREATES A LOT OF EXTRA ITEMS!
   upgrade          upgrade MySQL database schema to current version
+  dump [filename]  backup the current database to file (without .tar.gz suffix)
```
Comment (eldering) [inline sql/dj_setup_database.in:38]: I think it should be `sql.gz`

Comment (vmcj) [inline sql/dj_setup_database.in:38]:
```suggestion
  dump [filename]  backup the current database to file (without .sql.gz suffix)
```

**Review by eldering — APPROVED** [review]

**Pull request merged**